Update data-tests.md (clarify usage of singular tests in the `tests` directory) (#6780)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/data-tests.md

The name of this test is the name of the file: `assert_total_payment_amount_is_positive`. 

Note:

> - you won't need to include semicolons (;) at the end of the SQL statement in your singular test files as it can cause your test to fail.

Reviewer: ```suggestion
- Omit semicolons (;) at the end of the SQL statement in your singular test files, as they can cause your test to fail.
```

> - Singular tests placed in the tests directory are automatically executed when running dbt test. Do not reference singular tests in `model_name.yml`, as they are not treated as generic tests or macros, and doing so will result in an error.

Reviewer: ```suggestion
- Singular tests placed in the tests directory are automatically executed when running `dbt test`. Don't reference singular tests in `model_name.yml`, as they are not treated as generic tests or macros, and doing so will result in an error.
```
